vllm/multimodal/registry.py

```diff
@@ -105,41 +116,40 @@ def register_input(
 
         See :meth:`MultiModalPlugin.register_input_processor` for more details.
         """
-        return self._get_plugin_for_data_type(data_type) \
+        return self._get_plugin_for_internal_data_type(data_type) \
             .register_input_processor(processor)
 
-    def register_image_pixel_input(
+    def register_image_input(
             self,
-            processor: Optional[
-                MultiModalInputProcessor[ImagePixelData]] = None):
+            processor: Optional[MultiModalInputProcessor[ImageData]] = None):
         """
         Register an input processor for image pixel data to a model class.
 
         See :meth:`MultiModalPlugin.register_input_processor` for more details.
         """
-        return self.register_input(ImagePixelData, processor)
-
-    def register_image_feature_input(
-        self,
-        processor: Optional[
-            MultiModalInputProcessor[ImageFeatureData]] = None):
-        """
-        Register an input processor for image feature data to a model class.
+        return self.register_input(ImageData, processor)
 
-        See :meth:`MultiModalPlugin.register_input_processor` for more details.
-        """
-        return self.register_input(ImageFeatureData, processor)
-
-    def process_input(self, data: MultiModalData, model_config: ModelConfig,
+    def process_input(self, data: Union[MultiModalData, Dict[str, Any]],
+                      model_config: ModelConfig,
                       vlm_config: VisionLanguageConfig):
         """
-        Apply an input processor to a :class:`~MultiModalData` instance passed
-        to the model.
+        Apply an input processor before passing in to the model.
+
+        If the data is internally supplied (for profiling), 
+        it's of type :class:`~MultiModalData`.
+        If externally supplied through user API, it's of type dict. 
         
         See :meth:`MultiModalPlugin.process_input` for more details.
         """
-        return self._get_plugin_for_data_type(type(data)) \
-            .process_input(data, model_config, vlm_config)
+        if isinstance(data, MultiModalData):
+            return self._get_plugin_for_internal_data_type(type(data)) \
+                .process_input(data, model_config, vlm_config)
+        else:
+            result_list = [
+                self._process_external_input(d, model_config, vlm_config)
```

Comment:
I think we should process the inputs based on the key ("image") in addition to the value.

Reply:
sounds good, added.